src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -3436,5 +3455,944 @@ public void Deconstruct(out int a, out int b)
+
+            var x3 = GetDeconstructionVariable(tree, "x3");
+            var x3Symbol = model.GetDeclaredSymbol(x3);
+            var x3Ref = GetReference(tree, "x3");
+            Assert.Equal("System.Byte Script.x3", x3Symbol.ToTestDisplayString());
+            VerifyModelForDeconstructionField(model, x3, x3Ref);
+
+            // extra check on var
+            var x123Var = (TypedVariableComponentSyntax)x1.Parent.Parent;
+            Assert.Equal("var", x123Var.Type.ToString());
+            Assert.Null(model.GetSymbolInfo(x123Var.Type).Symbol); // The var in `var (x1, x2)` has no symbol
+        }
+
+        [Fact]
+        public void VarTypeInVarDeconstructionInScript()
+        {
+            var source =
+@"
+class var
+{
+    public static implicit operator var(int i) { return null; }
+}
+var (x1, (x2, x3)) = (1, (2, 3));
+System.Console.Write($""{x1} {x2} {x3}"");
+";
+
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics(
+                // (6,6): error CS8136: Deconstruction 'var (...)' form disallows a specific type for 'var'.
+                // var (x1, (x2, x3)) = (1, (2, 3));
+                Diagnostic(ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, "x1").WithLocation(6, 6),
+                // (6,11): error CS8136: Deconstruction 'var (...)' form disallows a specific type for 'var'.
+                // var (x1, (x2, x3)) = (1, (2, 3));
+                Diagnostic(ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, "x2").WithLocation(6, 11),
+                // (6,15): error CS8136: Deconstruction 'var (...)' form disallows a specific type for 'var'.
+                // var (x1, (x2, x3)) = (1, (2, 3));
+                Diagnostic(ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, "x3").WithLocation(6, 15)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var x1 = GetDeconstructionVariable(tree, "x1");
+            var x1Symbol = model.GetDeclaredSymbol(x1);
+            var x1Ref = GetReference(tree, "x1");
+            Assert.Equal("Script.var Script.x1", x1Symbol.ToTestDisplayString());
+            VerifyModelForDeconstructionField(model, x1, x1Ref);
+
+            var x3 = GetDeconstructionVariable(tree, "x3");
+            var x3Symbol = model.GetDeclaredSymbol(x3);
+            var x3Ref = GetReference(tree, "x3");
+            Assert.Equal("Script.var Script.x3", x3Symbol.ToTestDisplayString());
+            VerifyModelForDeconstructionField(model, x3, x3Ref);
+
+            // extra check on var
+            var x123Var = (TypedVariableComponentSyntax)x1.Parent.Parent;
+            Assert.Equal("var", x123Var.Type.ToString());
+            Assert.Null(model.GetSymbolInfo(x123Var.Type).Symbol); // The var in `var (x1, x2)` has no symbol
```

Comment:
Please add check for GetAliasInfo

---
In reply to: [80523960](https://github.com/dotnet/roslyn/pull/13957#discussion_r80523960) [](ancestors = 80523960)